server/controllers/profile.js

```diff
@@ -20,12 +22,29 @@ exports.getProfileFromUserId = asyncHandler(async (req, res) => {
     throw new Error("User info is not correct");
   }
 });
-const { profile } = require("console");
 
 // @route GET /profiles
-// @desc get all profiles
+// @desc get all profiles and if query string is added, then sort according to the included parameters
 // @access Public
 exports.loadProfiles = asyncHandler(async (req, res, next) => {
+  let { address, dropInDate, dropOffDate } = req.query;
+  if (address || dropInDate || dropOffDate) {
+    [address, dropInDate, dropOffDate] = removeWhiteSpace([
+      address,
+      dropInDate,
+      dropOffDate,
+    ]);
+
+    let profiles = await Profile.find({
+      address: { $regex: address || "", $options: "i" },
```

Comment:
Optional: While querying for sitter profiles, it would be nice if we could filter out the logged in user :)

Ref: https://docs.mongodb.com/manual/reference/operator/query/ne/